app/models/redis_hit_counter.rb

```diff
@@ -0,0 +1,172 @@
+# A class for keeping track of hits/IP addresses in redis. Writes the values in
+# redis to the database when you call save_recent_counts.
+class RedisHitCounter
+  # Records a hit for the given IP address on the given work ID. If the IP
+  # address hasn't visited the work within the current 24 hour block, we
+  # increment the work's hit count. Otherwise, we do nothing.
+  def add(work_id, ip_address)
+    timestamp = current_timestamp
+    key = "#{work_id}:#{timestamp}"
+
+    # Simultaneously add the IP address to the set for this work/date combo,
+    # and add the key we're using to the hash mapping from keys to timestamps.
+    added, _ = redis.multi do |multi|
```

Comment:
https://rubystyle.guide/#trailing-underscore-variables

We should be able to avoid the warning by naming the second variable `_something`.

Reply:
Oh, good call!

(I disagree with the style cop there, because I feel like omitting the underscore makes it less clear that it's doing a parallel assignment. It's easy to miss a single comma if you're not paying attention. Having a variable name prefixed with an underscore is a good compromise.)